src/mss/windows.py

```diff
@@ -254,5 +295,66 @@ def _grab_impl(self, monitor: Monitor, /) -> ScreenShot:
-        return None
+        """Retrieve all cursor data. Pixels have to be RGB.
+
+        [1] user32.DrawIcon(HDC(memdc), 0, 0, hcursor)
+        Sometimes the memdc value is greater than the 32 bit limit
+        and that results in
+        'ctypes.ArgumentError: argument 1: OverflowError: int too long to convert'
+        but casting it to HDC type seems to fix the issue.
+
+        [2] user32.GetIconInfo(hcursor, self._handles.icon_info)
+        GetIconInfo also returns the handle for mask bitmap and the handle for color bitmap
+        but the color bitmap handle is null in case of monochrome cursors.
+
+        [3] is_monochrome = self._handles.icon_info.hbmColor is None
+        The correct way to detect monochrome cursors seems to be a unique property of their
+        mask bitmap. The height of the mask bitmap of a monochrome cursor is twice its width
+        (https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-iconinfo)
+        But I cannot find the correct way of getting a bitmap's dimensions, therefore I
+        just eneded up checking if the color bitmap is null.
+
+        [4]
+        The data received using DrawIcon is in the format BGRA but in case of monochrome
+        cursors the alpha value of every pixel is 0 for some reason. Therefore, the alpha
+        value of every non black pixel has to be manually set to 255.
+        """
+        srcdc, memdc = self._handles.srcdc, self._handles.memdc
+        gdi, user32 = self.gdi32, self.user32
+        width, height = 32, 32
+        user32.GetCursorInfo(self._handles.cursor_info)
+        hcursor = self._handles.cursor_info.hCursor
+        pos_screen = self._handles.cursor_info.ptScreenPos
+
+        if self._handles.region_width_height != (width, height):
+            self._handles.region_width_height = (width, height)
+            self._handles.bmi.bmiHeader.biWidth = width
+            self._handles.bmi.bmiHeader.biHeight = -height
+            self._handles.data = ctypes.create_string_buffer(width * height * 4)
+            if self._handles.bmp:
+                gdi.DeleteObject(self._handles.bmp)
+            self._handles.bmp = gdi.CreateCompatibleBitmap(srcdc, width, height)
+            gdi.SelectObject(memdc, self._handles.bmp)
+
+        user32.DrawIcon(HDC(memdc), 0, 0, hcursor)  # Why HDC? [1]
+        bits = gdi.GetDIBits(memdc, self._handles.bmp, 0, height, self._handles.data, self._handles.bmi, DIB_RGB_COLORS)
+        if bits != height:
+            raise ScreenShotError("gdi32.GetDIBits() failed.")
+
+        user32.GetIconInfo(hcursor, self._handles.icon_info)  # [2]
+        is_monochrome = self._handles.icon_info.hbmColor is None  # [3]
+        ratio = ctypes.windll.shcore.GetScaleFactorForDevice(0) / 100
+        region = {
+            "left": round(pos_screen.x * ratio - self._handles.icon_info.xHotspot),
+            "top": round(pos_screen.y * ratio - self._handles.icon_info.yHotspot),
```

Comment:
Hi, I used this PR in my project and from my testing multiplying by ratio actually messes things up. When i removed it everything worked fine even on strange DPI scalings in multiple different positions of multi monitor setup. For testing I used https://github.com/pavlobu/deskreen with https://www.amyuni.com/forum/viewtopic.php?t=3030